decision-records/0004-data-api.md

```diff
@@ -0,0 +1,406 @@
+# Expose versions data through the collection API
+
+- Date: 2023-10-25
+
+## Context and Problem Statement
+
+### Context
+
+The need to extend the current API comes as part of our collaboration with Terms of Service; Didn't Read (ToS;DR), whose web application will be adapted to obtain data from public Open Terms Archive collections instead of the ToS;DR server database.
+
+The terms annotated by ToS;DR are currently taken from original web documents stored on the ToS;DR server database. Open Terms Archive provides a significantly higher quality source for terms, for example by cleaning up all content that is not related to the terms, combining documents scattered across several pages, or normalising PDF documents into Markdown.
+
+To allow the transition from historical ToS;DR's crawler and database to the publicly accessible Open Terms Archive collections, ToS;DR requires access to the content of the versions. Datasets are not satisfactory as they are produced on a weekly basis, whereas a workflow supporting annotation of just-added terms is required by ToS;DR. This RFC outlines possible implementations of an API providing this data.
+
+## Proposed solutions
+
+### Solution A
+
+#### Base URL
+
+`<collection host>/api/:version`
+
+#### Endpoint
+
+##### `GET /version/:serviceId/:termsType/:date.md`
+
+Return the Markdown content of the version of the terms that was applicable at the given date
+
+##### Parameters
+
+| Parameter | Type   | Description            |
+| --------- | ------ | ---------------------- |
+| serviceId | URL-encoded string | The ID of the service |
+| termsType | URL-encoded string | The name of terms type |
+| date | URL-encoded ISO 8601 datetime string | The date and time for which the version is requested |
+
+Note about the date:
+- A full date and time is required, and not a simple date (such as `2023-10-24`), to avoid ambiguities on days where a version changed, and timezone differences between client and server.
```

Comment:
Maybe pin the timezone to UTC? So "A full UTC data and time"?

Reply:
If the timezone is defined as an offset from UTC (ex: `2023-12-06T12:34:56+02:00`) is still ISO 8601 valid. 
So, why forcing the timezone to be UTC?